akka-stream/src/main/scala/akka/stream/impl/fusing/StreamOfStreams.scala

```diff
@@ -218,3 +225,191 @@ private[fusing] object StreamOfStreams {
     }
   }
 }
+
+/**
+ * INTERNAL API
+ */
+object PrefixAndTail {
+
+  sealed trait MaterializationState
+  case object NotMaterialized extends MaterializationState
+  case object AlreadyMaterialized extends MaterializationState
+  case object TimedOut extends MaterializationState
+
+  case object NormalCompletion extends MaterializationState
+  case class FailureCompletion(ex: Throwable) extends MaterializationState
+
+  trait TailInterface[T] {
+    def pushSubstream(elem: T): Unit
+    def completeSubstream(): Unit
+    def failSubstream(ex: Throwable)
+  }
+
+  final class TailSource[T](
+    timeout: FiniteDuration,
+    register: TailInterface[T] ⇒ Unit,
+    pullParent: Unit ⇒ Unit,
+    cancelParent: Unit ⇒ Unit) extends GraphStage[SourceShape[T]] {
+    val out: Outlet[T] = Outlet("Tail.out")
+    val materializationState = new AtomicReference[MaterializationState](NotMaterialized)
+    override val shape: SourceShape[T] = SourceShape(out)
+
+    private final class TailSourceLogic(_shape: Shape) extends GraphStageLogic(_shape) with OutHandler with TailInterface[T] {
+      setHandler(out, this)
+
+      override def preStart(): Unit = {
+        materializationState.getAndSet(AlreadyMaterialized) match {
+          case AlreadyMaterialized ⇒
+            failStage(new IllegalStateException("Tail Source cannot be materialized more than once."))
+          case TimedOut ⇒
+            // Already detached from parent
+            failStage(new TimeoutException(s"Tail Source has not been subscribed in $timeout."))
```

Comment:
i.e. I’m currently telling you how to fix that failure, unless I’m missing something